README.md
# Sensi-ATM-web
Web demo of ATM authentication with sensipass app


mysensi/changelog.txt
21/01/2016 - QR code reader login demo.
22/01/2016 - js sucks



mysensi/qrcode/js/jsony_test.js
/**
var channelObj = JSON.parse("channel");
var myFeedsyShort = [];

for (var i=0;j<channelObj.feeds.lenght; i++){
	myFeedsyShort[i] = channelObj.feeds[i].created_at; (albo inne pole s feeds: entry_id, field1, field2	
}
**/
//-------------------------------------------
//var channelObj = JSON.parse("channel_json_string");

        

//var scriptTag = document.createElement('CHANNEL');
//scriptTag.src = "https://api.thingspeak.com/channels/72126/feeds.json?results=20";
//document.getElementsByTagName('HEAD')[0].appendChild(scriptTag);

    

//var channelObj = JSON.parse(Get(https://api.thingspeak.com/channels/72126/feeds.json?results=20));
function getAll(){
	
	function Get(yourUrl){
		var Httpreq = new XMLHttpRequest(); // a new request
		Httpreq.open("GET",yourUrl,false);
		Httpreq.send(null);
	return Httpreq.responseText; 
	}
	//-------------------------
	//to teraz array obektów z polami:
var channelObj = JSON.parse(Get(https://api.thingspeak.com/channels/72126/feeds.json?results=20));

//nowy, lepszy konstruktor
function feed(created, id, f1, f2){
	var createdAt = created;
	var entryId = id;
	var field1 = f1;
	var field2 = f2;
	
	this.getCreatedAt = finction(){
		return createdAt;
	};
	this.getEntryId = finction(){
		return entryId;
	};
	this.getField1 = finction(){
		return field1;
	};
	this.getfield2 = finction(){
		return field2;
	};
}
	
	var myFeedsyArr = []; 

	for (var i=0;j<channelObj.feeds.lenght; i++){
		myFedesyArr[i] = new feed(channelObj.feeds[i].created_at, channelObj.feeds[i].entry_id, channelObj.feeds[i].field1, channelObj.feeds[i].field2);	
	}
	return myFeedsyArr;
}



mysensi/qrcode/js/camera.js
var camera = (function(p_vid_id, p_inter, p_scale) {

	if (p_vid_id == undefined) {
		console.log("ERROR: You need to specify the id of the <video> element with the camera data stream.");
		return;
	}

	var vid_id   = p_vid_id;
	var interval = p_inter != undefined ? p_inter : 100;
	var scale    = p_scale != undefined ? p_scale : 0.5;

    var video    = document.getElementById(vid_id);
    var int_id   = null;

	function start() {
		int_id = setInterval(function(video, scale) { capture() }, interval);
	}

	function stop() {
		console.log("Clearing interval with id "+int_id);
		clearInterval(int_id);
	}

	function capture() {
		
		// console.time('capture');
	    var w = video.videoWidth * scale;
	    var h = video.videoHeight * scale;
	    var qr_can = document.getElementById('qr-canvas').getContext('2d');
	        qr_can.drawImage(video, 0, 0, w, h);
	    try        { qrcode.decode();  }
	    catch(err) { 
	    	$("#qr-value").text(err); 

	    	if (err != "Couldn't find enough finder patterns"){
	    			$("#tickGif").hide();
	    			$("#xGif").show();
	    			//window.setTimeout(fail, 1000);   // Arek
					fail();
	    	}

	    }
		// console.timeEnd('capture');
	} 
	
	// Arek
	function fail()  { setInterval(function() { window.location = 'unsuccessful.html' ; }, 100); }

	return {
		interval:interval,
		scale:scale,
		start:start,
		stop:stop,
		capture:capture
	}

})

mysensi/qrcode/js/Backup-init.js
/* JavaScript code */

navigator.getUserMedia = navigator.webkitGetUserMedia || navigator.mozGetUserMedia || navigator.msGetUserMedia;

var cam_video_id = "camsource"
	

window.addEventListener('DOMContentLoaded', function() {
    // Assign the <video> element to a variable
    var video = document.getElementById(cam_video_id);
    var options = {
        "audio": false,
        "video": {	"mandatory": {
    				"width": { "min": 640 },
    				"height": { "min": 480 }
  				},
  			"optional": [
    			{ "width": 1024 },
    			{ "width": { "min": 1024 }},
    			{ "width": { "max": 1024 }},
  			]
  		}
  		
    };
    var constraints = {
  			"audio": false,
  			"video": {
    			"width": { "min": 1024, "max": 1024 },
    			"height": { "min": 720, "max": 720 },
    			"miniFramerate": 30,
    			"require": ["width", "height"] // needed pre-Firefox 38 (non-spec)
  			},
		};
    
    
    // Replace the source of the video element with the stream from the camera
    if (navigator.getUserMedia) {
        navigator.getUserMedia(options, function(stream) {
            video.src = (window.URL && window.URL.createObjectURL(stream)) || stream;
        }, function(error) {
            console.log(error)
        });
        // Below is the latest syntax. Using the old syntax for the time being for backwards compatibility.
        // navigator.getUserMedia({video: true}, successCallback, errorCallback);
    } else {
        $("#qr-value").text('Sorry, native web camera streaming (getUserMedia) is not supported by this browser...')
        return;
    }
}, false);

$(document).ready(function() {
    if (!navigator.getUserMedia) return;
    cam = camera(cam_video_id);
    cam.start()
})

mysensi/qrcode/js/init.js
/* JavaScript code */

navigator.getUserMedia = navigator.webkitGetUserMedia || navigator.mozGetUserMedia || navigator.msGetUserMedia;

var cam_video_id = "camsource"
	

window.addEventListener('DOMContentLoaded', function() {
    // Assign the <video> element to a variable
    var video = document.getElementById(cam_video_id);
	
	var options = {
       "audio": false,
       "video": true
   };
	/*
	var options = {
       "audio": false,
       "video": {
           optional: [
               {minWidth: 320},
               {minWidth: 640},
               {minWidth: 800},
               {minWidth: 900},
               {minWidth: 1024},
               {minWidth: 1280},
               {minWidth: 1920},
               {minWidth: 2560}
           ];
       }
   };
   
   
    var options = {
        "audio": false,
        "video": {	"mandatory": {
    				"width": { "min": 640 },
    				"height": { "min": 480 }
  				},
  			"optional": [
    			{ "width": 1024 },
    			{ "width": { "min": 1024 }},
    			{ "width": { "max": 1024 }},
				{ "height": 720 },
    			{ "height": { "min": 720 }},
    			{ "height": { "max": 720 }},
				
  			]
  		}
    };
	*/
    var constraints = {
  			"audio": false,
  			"video": {
    			"width": { "min": 1024, "max": 1024 },
    			"height": { "min": 720, "max": 720 },
    			"miniFramerate": 30,
    			"require": ["width", "height"] // needed pre-Firefox 38 (non-spec)
  			}
		};
    
    
    // Replace the source of the video element with the stream from the camera
    if (navigator.getUserMedia) {
        navigator.getUserMedia(options, function(stream) {
            video.src = (window.URL && window.URL.createObjectURL(stream)) || stream;
        }, function(error) {
            console.log(error)
        });
        // Below is the latest syntax. Using the old syntax for the time being for backwards compatibility.
        // navigator.getUserMedia({video: true}, successCallback, errorCallback);
    } else {
        $("#qr-value").text('Sorry, native web camera streaming (getUserMedia) is not supported by this browser...')
        return;
    }
}, false);

$(document).ready(function() {
    if (!navigator.getUserMedia) return;
    cam = camera(cam_video_id);
    cam.start()
})

/*
stare moje optional:
var options = {
        "audio": false,
        "video": {	"mandatory": {
    				"width": { "min": 640 },
    				"height": { "min": 480 }
  				},
  			"optional": [
    			{ "width": 1024 },
    			{ "width": { "min": 1024 }},
    			{ "width": { "max": 1024 }}
  			]
  		}
    };
	*/